src/webcore/executor.rs

```diff
@@ -0,0 +1,226 @@
+// This file implements a futures-compatible executor which schedules futures
+// onto the JavaScript event loop. This implementation assumes there is a
+// single thread and is *not* compatible with multiple WebAssembly workers sharing
+// the same address space.
+
+use futures::future::{Future, ExecuteError, Executor};
+use futures::executor::{self, Notify, Spawn};
+use futures::Async;
+use std::collections::VecDeque;
+use std::result::Result as StdResult;
+use std::cell::{Cell, RefCell};
+use std::rc::Rc;
+use webcore::try_from::TryInto;
+use webcore::value::Reference;
+
+
+// Initial capacity of the event queue
+const INITIAL_QUEUE_CAPACITY: usize = 10;
+// Iterations to wait before allowing the queue to shrink
+const QUEUE_SHRINK_DELAY: usize = 25;
```

Comment:
There should be a `TODO` which says that we should test for the optimal sizes for these two constants.